src/model/core/EthqlStorage.ts

```diff
@@ -0,0 +1,90 @@
+import Web3 = require('web3');
+import { EthqlContext } from '../EthqlContext';
+
+interface PathObject {
+  query: string;
+  storageType: string;
+}
+
+class EthqlStorage {
+  private address: string;
+  private base: string;
+  private path: PathObject[];
+  private previousType: string;
```

Comment:
Using path[path.length - 1] would complicate things because the first query is not added to path but rather is used to update base. From then on queries are added to base and reference the previousType. It would produce index out of bounds on the second query.

Reply:
Ok, thanks for the explanation!